lms/static/sass/shared/_footer-edx.scss

```diff
@@ -19,6 +19,92 @@ footer#footer-edx-v3 {
   padding: $baseline*0.75 0;
   border-top: 1px solid $courseware-button-border-color;
 
+  .column-1 {
+    flex: 0 0 100%;
+    max-width: 100%;
+    position: relative;
+    width: 100%;
+    min-height: 1px;
+    padding-right: 15px;
+    padding-left: 15px;
+  }
+
+  .column-2 {
+    flex: 0 0 100%;
+    max-width: 100%;
+    position: relative;
+    width: 100%;
+    min-height: 1px;
+    padding-right: 15px;
+    padding-left: 15px;
+  }
+
+  .column-3 {
+    flex: 0 0 50%;
+    max-width: 50%;
+    position: relative;
+    width: 100%;
+    min-height: 1px;
+    padding-right: 15px;
+    padding-left: 15px;
+  }
+
+  .column-2-1 {
+    flex: 0 0 100%;
+    max-width: 100%;
+    position: relative;
+    width: 100%;
+    min-height: 1px;
+    padding-right: 15px;
+    padding-left: 15px;
+    //flex-basis: auto;
+  }
+
+  .column-2-2 {
+    flex: 0 0 100%;
+    max-width: 100%;
+    position: relative;
+    width: 100%;
+    min-height: 1px;
+    padding-right: 15px;
+    padding-left: 15px;
+    //flex-basis: auto;
+  }
+
+  .column-3 {
+    flex: 0 0 100%;
+    max-width: 100%;
+    position: relative;
+    width: 100%;
+    min-height: 1px;
+    padding-right: 15px;
+    padding-left: 15px;
+  }
+
+  .column-3-1 {
+    flex: 0 0 100%;
+    max-width: 100%;
+    position: relative;
+    width: 100%;
+    min-height: 1px;
+    padding-right: 15px;
+    padding-left: 15px;
+  }
+
+  .column-3-2 {
+    flex: 0 0 100%;
+    max-width: 100%;
+    position: relative;
+    width: 100%;
+    min-height: 1px;
+    padding-right: 15px;
+    padding-left: 15px;
+  }
```

Comment:
There are 8 classes that all share the same styles, can the approach be updated to be more DRY?

Reply:
This is particularly because the columns changes for every screen so we have to repeat which changes. but yes if there are no changes i am removing the repetition of those classes. And i am treating my columns in three different parts of screen. Their widths changes after every screen.

Reply:
@hamzamunir7300 regarding the place you were asking for more details about earlier, this is what I had in mind;
```
.column-styles {
  flex: 0 0 100%;
  max-width: 100%;
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: 15px;
  padding-left: 15px;

  &.column-3 {
    flex: 0 0 50%;
    max-width: 50%;
  }
}
```

Reply:
It looked like the only column what wasn't set to 100% was ```.column-3```, so I was suggesting just defaulting column-styles to 100%, and updating as appropriate.